Lib/zipimport.py

```diff
@@ -517,8 +517,9 @@ def _read_directory(archive):
                             num_extra_values = (len(extra_data) - 4) // 8
                             if num_extra_values > 3:
                                 raise ZipImportError(f"can't read header extra: {archive!r}", path=archive)
-                            values = struct.unpack_from(f"<{min(num_extra_values, 3)}Q",
-                                                        extra_data, offset=4)
+                            import struct
+                            values = list(struct.unpack_from(f"<{min(num_extra_values, 3)}Q",
```

Comment:
The new test caught the need for the list here. Prior to the fix you'd see:
```
ERROR: testZip64LargeFile (test.test_zipimport.UncompressedZipImportTestCase.testZip64LargeFile)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<frozen importlib._bootstrap_external>", line 1510, in _path_importer_cache
KeyError: '/home/jsirois/dev/python/cpython/build/test_python_worker_140690æ/junk95142.zip'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jsirois/dev/python/cpython/Lib/test/test_zipimport.py", line 89, in wrapper
    return fun(*args, **kwargs)
           ~~~^^^^^^^^^^^^^^^^^
  File "/home/jsirois/dev/python/cpython/Lib/test/test_zipimport.py", line 89, in wrapper
    return fun(*args, **kwargs)
           ~~~^^^^^^^^^^^^^^^^^
  File "/home/jsirois/dev/python/cpython/Lib/test/test_zipimport.py", line 849, in testZip64LargeFile
    self.doTestWithPreBuiltZip(".py", "large_file")
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/dev/python/cpython/Lib/test/test_zipimport.py", line 157, in doTestWithPreBuiltZip
    mod = importlib.import_module(".".join(modules))
          ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/dev/python/cpython/Lib/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1322, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1262, in _find_spec
  File "<frozen importlib._bootstrap_external>", line 1553, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1525, in _get_spec
  File "<frozen importlib._bootstrap_external>", line 1512, in _path_importer_cache
  File "<frozen importlib._bootstrap_external>", line 1488, in _path_hooks
  File "<frozen zipimport>", line 98, in __init__
  File "<frozen zipimport>", line 528, in _read_directory
AttributeError: 'tuple' object has no attribute 'pop'
```

Reply:
another good catch! glad you added the test case!